### ISSUE TITLE
API Documentation: Doxygen

### ISSUE THREAD

**Issue by ovflowd** [sant0ro/Yupi#136]
Planning to use [Doxygen](http://www.doxygen.org/) for API and Code Documentation.

**Comment by TheDoct0r11** [comment]
You're talking about the rest api? That isn't Yaml, that's either JSON or XML.

**Comment by ovflowd** [comment]
No, i'm talking about the API Documentation. Swagger uses YAML to generate the API Documentation.
And Swagger is used for API Documentation, can also used for the REST Documentation. But mainly used for the Classes/Code Documentation xD

**Comment by TheDoct0r11** [comment]
I can't see anything on the Swagger website not related to REST.
Also why would one want to write the docs within an extra file?

I'd rather suggest using a tool that uses the documentation that is within the code.

[Doxygen](http://www.doxygen.org) is a very common tool.

**Comment by ovflowd** [comment]
@BurakDev can do that. With Doxygen

**Comment by ovflowd** [comment]
@lai0n will do the Documentation with Doxygen.

**Comment by ovflowd** [comment]
**Remember.:** need to do the HTML API Documentation and In-Code Documentation.
